src/recycler.rs

```diff
@@ -184,7 +184,28 @@ mod tests {
             });
             ThreadNanny { _hdl };
         }
+    }
 
+    #[test]
+    fn test_window() {
+        let recycler: Recycler<Foo> = Recycler::default();
+        let mut window = vec![None];
+        let (sender, receiver) = channel();
+        {
+            // item is in the window while its in the pipeline
+            // which is used to serve requests from other threads
+            let mut item = recycler.allocate();
+            item.as_mut().x = 1;
+            window[0] = Some(item);
```

Comment:
@aeyakovenko, This usage doesn't have anything to do with recycling. `Arc::new()` the recyclable before adding it to the window, and then clone the Arc as you send it. It'll live till both the window and receiving thread drop it, and then return to the recycler.